app/Http/Middleware/JsonResponse.php

```diff
@@ -11,7 +11,12 @@ class JsonResponse
 {
     public function handle(Request $request, Closure $next): mixed
     {
-        $request->headers->set('Accept', 'application/json');
+        // Don't override the Accept header if it's already set to the media type
+        // required by JSON:API responses.
+        $currentAccept = $request->header('Accept');
+        if ($currentAccept !== 'application/vnd.api+json') {
+            $request->headers->set('Accept', 'application/json');
+        }
```

Comment:
Also, what is this block doing? The comment suggests it forces `application/json` if `application/vnd.api+json` is not provided, but if I pass in something like `application/image` (or even `application/json`), I get back an error stating it's not a valid type for the request.